google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java

```diff
@@ -155,9 +156,44 @@ public static TransactionOption commitStats() {
    * process in the commit phase (when any needed locks are acquired). The validation process
    * succeeds only if there are no conflicting committed transactions (that committed mutations to
    * the read data at a commit timestamp after the read timestamp).
+   *
+   * @deprecated Use {@link Options#readLockMode(ReadLockMode)} instead.
    */
+  @Deprecated
   public static TransactionOption optimisticLock() {
-    return OPTIMISTIC_LOCK_OPTION;
+    return Options.readLockMode(ReadLockMode.OPTIMISTIC);
+  }
+
+  /**
+   * Returns a {@link TransactionOption} to set the desired {@link ReadLockMode} for a read-write
+   * transaction.
+   *
```

Comment:
Can we also add some generic advice when to use what? (Otherwise, I think that most users will automatically assume that 'optimistic is always better than pessimistic').

So for example something like this:

```suggestion
   * Optimistic locking can improve throughput, as transactions wait with taking locks until they try to commit,
   * but increases the probability that Spanner needs to abort a transaction due to lock conflicts. Pessimistic
   * locking instructs transactions to take locks earlier, which means that concurrent transactions might have
   * to wait for other transactions to release locks before proceeding. Pessimistic locking reduces the probability
   * that Spanner needs to abort a transaction due to lock conflicts.
```

Reply:
I initially added some [advice](https://github.com/googleapis/java-spanner/commit/1c38bc5e3849ac55e176a83f5c50f4ed3a6e3153#diff-992d480774d874d97423da09934b0395ef460130558e2fd6f7b1461ba32535c0R173-R185) to the document but removed it because I felt it deserved a more detailed explanation in a separate link. I've now added the advice back, and we will update this document once the RR + Pessimistic feature is launched.